src/weather/open_meteo.rs

```diff
@@ -79,6 +81,7 @@ impl From<Response> for CurrentConditions {
             temp_f: result.current_weather.temperature,
             time_of_day: TimeOfDay::from(result.current_weather.is_day),
             icon: None,
+            provider: super::Provider::OpenMeteo,
```

Comment:
This feels like it should be part of the trait somehow, not hard-coded inline here.